packages/flutter_tools/lib/src/android/android_device.dart

```diff
@@ -366,7 +366,7 @@ class AndroidDevice extends Device {
 
   @override
   Future<bool> isAppInstalled(
-    AndroidApk app, {
+    ApplicationPackage app, {
```

Comment:
so for this call, will we just be more permissive and allow any ApplicationPackage?

Reply:
I wonder if the right solution to this is not to make Device:

```dart
abstract class Device<A extends ApplicationPackage> {
  isAppInstalled(A app, ...) {}
}
```

Reply:
That would be a much better solution, but it also means every single usage of the base class `Device` becomes `Device<ApplicationPackage>`, and we have hundreds of them.

`info: Specify type annotations. (always_specify_types)`

I gave up with about 400 more warnings to go:
https://github.com/flutter/flutter/commit/e780f1d47bc6412eb4a66b4b374462b2531d1ab4

Reply:
What do you want me to do with this PR?

Reply:
let me make one more pass, we can land as is